sql/core/src/test/scala/org/apache/spark/sql/FilterPushdownBenchmark.scala

```diff
@@ -0,0 +1,195 @@
+
+        Filter Pushdown (id IS NULL):            Best/Avg Time(ms)    Rate(M/s)   Per Row(ns)   Relative
+        ------------------------------------------------------------------------------------------------
+        Parquet Vectorized                            2081 / 2123          0.5        1984.4       1.0X
+        Parquet Vectorized (Pushdown)                   36 /   37         29.3          34.1      58.1X
+        Native ORC Vectorized                         1616 / 1645          0.6        1540.7       1.3X
+        Native ORC Vectorized (Pushdown)                41 /   43         25.7          39.0      50.9X
+        */
+
+        // Negative cases: Select all rows which means the predicate is always true.
+        /*
+        Java HotSpot(TM) 64-Bit Server VM 1.8.0_152-b16 on Mac OS X 10.13.2
+        Intel(R) Core(TM) i7-4770HQ CPU @ 2.20GHz
+
+        Filter Pushdown (id > -1):               Best/Avg Time(ms)    Rate(M/s)   Per Row(ns)   Relative
+        ------------------------------------------------------------------------------------------------
+        Parquet Vectorized                            8346 / 8516          0.1        7959.8       1.0X
+        Parquet Vectorized (Pushdown)                 8611 / 8630          0.1        8212.4       1.0X
+        Native ORC Vectorized                         7700 / 7940          0.1        7343.2       1.1X
+        Native ORC Vectorized (Pushdown)              7572 / 7635          0.1        7221.5       1.1X
+
+        Filter Pushdown (id != -1):              Best/Avg Time(ms)    Rate(M/s)   Per Row(ns)   Relative
+        ------------------------------------------------------------------------------------------------
+        Parquet Vectorized                            8088 / 8297          0.1        7713.2       1.0X
+        Parquet Vectorized (Pushdown)                 7110 / 8674          0.1        6780.8       1.1X
+        Native ORC Vectorized                         7430 / 7567          0.1        7086.0       1.1X
+        Native ORC Vectorized (Pushdown)              7739 / 7832          0.1        7380.9       1.0X
+
+        Filter Pushdown (id IS NOT NULL):        Best/Avg Time(ms)    Rate(M/s)   Per Row(ns)   Relative
+        ------------------------------------------------------------------------------------------------
+        Parquet Vectorized                            7927 / 8284          0.1        7560.3       1.0X
+        Parquet Vectorized (Pushdown)                 7329 / 7332          0.1        6989.6       1.1X
+        Native ORC Vectorized                         7928 / 7971          0.1        7560.5       1.0X
+        Native ORC Vectorized (Pushdown)              7392 / 7502          0.1        7049.9       1.1X
+        */
+        benchmark.run()
+      }
+    }
+  }
+
+  def main(args: Array[String]): Unit = {
+    // Positive cases: Select one or no rows
```

Comment:
Split the positive cases to multiple, as suggested below. We need to see the perf for different predicates.